x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/conditional/Case.java

```diff
@@ -224,18 +227,13 @@ public boolean foldable() {
             if (condition.condition.foldable() == false) {
                 return false;
             }
-            Object o = condition.condition.fold();
-            if (o instanceof List) {
+            if (Boolean.TRUE.equals(condition.condition.fold())) {
                 /*
                  * multivalued fields fold to null which folds to false.
                  * So they *are* foldable if the value is foldable.
```

Comment:
I think the comment here means the converse.
```suggestion
                 * So they *are* foldable if remaining list of conditions is foldable.
```
Same in line 271.